pkg/controllers/manila/manila.go

```diff
@@ -94,14 +95,23 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 		return nil
 	}
 
+	var err403 gophercloud.ErrDefault403
+	var err404 gophercloud.ErrDefault404
+	var errNoEndpoint *gophercloud.ErrEndpointNotFound
 	shareTypes, err := c.openStackClient.GetShareTypes()
 	if err != nil {
-		switch err.(type) {
-		case gophercloud.ErrDefault403:
+		switch {
+		case errors.As(err, &err403):
```

Comment:
you may want to use that IsNotFound wrapper for these as well

Reply:
You know, you right. You never know what to expect from gophercloud.